### PR TITLE
Properly rename config members.

### PR DESCRIPTION
**Pull request by spershin**
Summary:
The config members are quite confusing, renaming them makes
it easier to understand.

Differential Revision: D41708793

### CODE REVIEW

File: velox/exec/Driver.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ DriverCtx::DriverCtx(
       task(_task) {}
 
 const core::QueryConfig& DriverCtx::queryConfig() const {
-  return task->queryCtx()->config();
+  return task->queryCtx()->queryConfig();
```
Comment (xiaoxmeng) [inline velox/exec/Driver.cpp:41]: I don't think we need to rename config to queryConfig? as it is a member of queryCotx? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@spershin thanks for the refactoring!

File: velox/exec/Driver.cpp (hunk 1 above)
Comment (spershin) [reply velox/exec/Driver.cpp:41]: > I don't think we need to rename config to queryConfig? as it is a member of queryCotx? Thanks!

@xiaoxmeng, thanks for taking time to review the change.

TLDR: The config and queryConfig are very different objects.

More verbose:
QueryCtx is Context, which is BaseConfigManager, which has the following member:
```
std::shared_ptr<const Config> config_;
```
And that member actually possesses all the config settings.

While QueryCtx had this member:
```
QueryConfig config_;
```
Where QueryConfig does not actually have anything, but a simple wrapper around BaseConfigManager with a bunch of helper methods to fetch settings.

It was all messed up and hard to follow and confused me for quite a long time until I finally found some time to understand what is what in that mess.

I've just changed the names to reflect what they are. :)

Comment (xiaoxmeng) [reply velox/exec/Driver.cpp:41]: @spershin thanks for the explanation!

File: velox/core/QueryConfig.h [hunk 2]
```diff
@@ -360,14 +363,14 @@ class QueryConfig {
...
   }
   template <typename T>
   std::optional<T> get(const std::string& key) const {
-    return std::optional<T>(config_->get<T>(key));
+    return std::optional<T>(configManager_->get<T>(key));
   }
 
  private:
-  BaseConfigManager* config_;
+  BaseConfigManager* configManager_;
```
Comment (xiaoxmeng) [inline velox/core/QueryConfig.h:374]: Add folly non null annotation?

**Review by xiaoxmeng — APPROVED** [review]
@spershin LGTM. Thanks for the cleanup and explanation of the renaming reason!
